src/metabase/lib/equality.cljc

```diff
@@ -423,6 +425,14 @@
      ;; Prefer the more precise integer ID matches over the name-based ones.
      (merge by-name basic))))
 
+(def ^{:arglists '([needles haystack-metadatas]
+                   [needles haystack-metadatas opts]
+                   [query stage-number needles haystack-metadatas]
+                   [query stage-number needles haystack-metadatas opts])}
+  closest-matches-in-metadata
+  "The cached version of [[closest-matches-in-metadata*]]."
+  (memoize/lru closest-matches-in-metadata* :lru/threshold 8))
```

Comment:
- To fix #33676, we could cache `breakoutable-columns`, but that spends most of the time in this function and caching this seem more generally useful.
- I expect that eventually the siblings of this function will have to be cached.
- The number 8 is quite random, but I think it should be enough for an action on a page and still not way too much. A lower number would probably work too.
- A TTL cache might be more appropriate here, but I didn't want to spend time porting that too.